### ISSUE TITLE
Support 18F14K50

### ISSUE THREAD

**Issue by ghost** [signal11/m-stack#2]
Would be nice with support for 18F14K50

**Comment by moondeck** [comment]
Sorry for reviving an old issue, but does m-stack support 18F14K50 now?

**Comment by signal11** [comment]
I don't have one of those chips. Have you tried it on yours? There are some differences on that one. It only has 8 endpoints, and I don't know what else is different.

**Comment by moondeck** [comment]
I havent tried yet, i will be able to try tomorrow. I want to port an existing application, that is using the non-free Microchip stack, so that the full source can be open, is it compatible with it? I must say, i have no idea about USB, or how it works on PIC, i am kinda hoping it will be mostly painless to do.

**Comment by DavidEGrayson** [comment]
> using the non-free Microchip stack

Microchip's USB stack is under the Apache license the last time I checked; what do you mean?

**Comment by moondeck** [comment]
Turns out you were right @DavidEGrayson ! Its Apache 2.0! Wasnt aware of that, i am trying to port over Numato's FPGA on-board programmer software, so they can open-source the entire thing.
They told me that they cant because the stack from Microchip was not free, and this article also suggest so:
http://dangerousprototypes.com/blog/2011/08/30/editorial-our-friend-microchip-and-open-source/

**Comment by DavidEGrayson** [comment]
Yeah, I believe the license changed some time in the last few years.  They also made it so the USB stack supports XC8 instead of just C18.

**Comment by bayesiandog** [comment]
Sorry for reviving an old issue as well, but does m-stack support 18F14K50 now?

Has anyone ported it already? What needs to be done to do so?
